src/OldRod.Core/Recompiler/Transform/ArrayAccessTransform.cs

```diff
@@ -93,12 +95,14 @@ private static bool IsArrayMemberAccess(CilInstruction instruction, out IMethodD
         {
             if (instruction.OpCode.Code == CilCode.Call
                 && instruction.Operand is IMethodDefOrRef m
-                && m.DeclaringType.IsTypeOf("System", "Array"))
+                && (m.DeclaringType.IsTypeOf("System", "Array") 
+                    || (m.DeclaringType is TypeSpecification ts 
```

Comment:
Is loosening this check necessary? I cannot really remember seeing any sample that didn't use `System.Array` as declaring type in its member reference.

I fear this might introduce some false positives with "normal" calls to `Array.GetValue` etc.

Reply:
KoiVM uses SZArray signature as the base type for the `Address` member as seen here:
https://github.com/yck1509/KoiVM/blob/master/KoiVM/ILAST/Transformation/ArrayTransform.cs#L72-L80
This is why the check had to be modified.